src/H5FA.c

```diff
@@ -671,44 +687,42 @@ END_FUNC(PRIV) /* end H5FA_delete() */
  *
  *-------------------------------------------------------------------------
  */
-BEGIN_FUNC(PRIV, ERR, int, H5_ITER_CONT, H5_ITER_ERROR,
-           H5FA_iterate(H5FA_t *fa, H5FA_operator_t op, void *udata))
-
-    /* Local variables */
+int
+H5FA_iterate(H5FA_t *fa, H5FA_operator_t op, void *udata)
+{
     uint8_t *elmt = NULL;
     hsize_t  u;
-    int      cb_ret = H5_ITER_CONT; /* Return value from callback */
+    int      ret_value = H5_ITER_CONT;
 
-    /*
-     * Check arguments.
-     */
+    FUNC_ENTER_NOAPI(H5_ITER_ERROR)
+
+    /* Check arguments */
     HDassert(fa);
     HDassert(op);
     HDassert(udata);
 
     /* Allocate space for a native array element */
     if (NULL == (elmt = H5FL_BLK_MALLOC(fa_native_elmt, fa->hdr->cparam.cls->nat_elmt_size)))
-        H5E_THROW(H5E_CANTALLOC, "memory allocation failed for fixed array element")
+        HGOTO_ERROR(H5E_FARRAY, H5E_CANTALLOC, H5_ITER_ERROR,
+                    "memory allocation failed for fixed array element")
 
     /* Iterate over all elements in array */
-    for (u = 0; u < fa->hdr->stats.nelmts && cb_ret == H5_ITER_CONT; u++) {
+    for (u = 0; u < fa->hdr->stats.nelmts && ret_value == H5_ITER_CONT; u++) {
         /* Get array element */
         if (H5FA_get(fa, u, elmt) < 0)
-            H5E_THROW(H5E_CANTGET, "unable to delete fixed array")
+            HGOTO_ERROR(H5E_FARRAY, H5E_CANTGET, H5_ITER_ERROR, "unable to delete fixed array")
 
-        /* Make callback */
-        if ((cb_ret = (*op)(u, elmt, udata)) < 0) {
-            H5E_PRINTF(H5E_BADITER, "iterator function failed");
-            H5_LEAVE(cb_ret)
-        } /* end if */
-    }     /* end for */
-
-    CATCH
+        /* Invoke callback */
+        if ((ret_value = (*op)(u, elmt, udata)) < 0)
+            HGOTO_DONE(ret_value)
```

Comment:
Same comment about maybe using HGOTO_ERROR

Reply:
Same here:  HERROR w/break out of loop is typical for the rest of the library.

Reply:
Fixed